src/client/content_cao.cpp

```diff
@@ -1174,7 +1174,7 @@ void GenericCAO::step(float dtime, ClientEnvironment *env)
 	}
 
 	if (!getParent() && m_prop.automatic_face_movement_dir &&
-			(fabs(m_velocity.Z) > 0.001 || fabs(m_velocity.X) > 0.001)) {
+			(fabs(m_velocity.Z) > 0.001f || fabs(m_velocity.X) > 0.001f)) {
```

Comment:
In the end, the floating point registers are internally 80-bit wide anyway. There's no benefit to append the `f`.

Reply:
> There's no benefit to append the `f`.

It's cleaner to append the 'f', so you can see that's a float and not a double at the first sight.

Reply:
Checking the disassembly, gcc 10.2 generates an extra "Convert Scalar Single-Precision Floating-Point Value to Scalar Double-Precision Floating-Point Value" instruction when using `0.001` over `0.001f`:
```diff
 	c5 f8 54 c1          	vandps xmm0,xmm0,xmm1
-	c5 f8 2e 05 00 00 00 00 	vucomiss xmm0,DWORD PTR [rip+0x0]        # <_ZN10GenericCAO4stepEfP17ClientEnvironment+0xdcb>
-	0f 86 af 00 00 00    	jbe    <_ZN10GenericCAO4stepEfP17ClientEnvironment+0xe80>
+	c5 fa 5a c0          	vcvtss2sd xmm0,xmm0,xmm0
+	c5 f9 2e 05 00 00 00 00 	vucomisd xmm0,QWORD PTR [rip+0x0]        # <_ZN10GenericCAO4stepEfP17ClientEnvironment+0xdcf>
+	0f 86 af 00 00 00    	jbe    <_ZN10GenericCAO4stepEfP17ClientEnvironment+0xe84>
 	49 8b 06             	mov    rax,QWORD PTR [r14]
```

Reply:
So you should be caring about unsigned values that should be `U`´d or ˋUL`’d if there’s one instruction more...

Reply:
I think gcc can convert signed constants to unsigned but it is not allowed to do so for double -> float since it loses precision.